plutus-metatheory/test/TestSimple.hs

```diff
@@ -29,71 +29,61 @@ succeedingEvalTests = ["succInteger"
   writeFile tmp example
-  putStrLn $ "test: " ++ test ++ " [" ++ command ++ "]"
-  withArgs ([command,"--input",tmp] ++ blah mode)  M.main
+  putStrLn $ "test: " ++ test ++ " [" ++ show command ++ "]"
+  withArgs (mkArgs tmp command) M.main
 
-runSucceedingTests :: String -> String -> [String] -> IO ()
-runSucceedingTests command mode [] = return ()
-runSucceedingTests command mode (test:tests) = catch
-  (runTest command mode test)
+runSucceedingTests :: Command -> [String] -> IO ()
+runSucceedingTests command [] = return ()
+runSucceedingTests command (test:tests) = catch
+  (runTest command test)
   (\case
     ExitFailure _ -> exitFailure
-    ExitSuccess   -> runSucceedingTests command mode tests)
+    ExitSuccess   -> runSucceedingTests command tests)
 
-runFailingTests :: String -> String -> [String] -> IO ()
-runFailingTests command mode [] = return ()
-runFailingTests command mode (test:tests) = catch
-  (runTest command mode test)
+runFailingTests :: Command -> [String] -> IO ()
+runFailingTests command [] = return ()
+runFailingTests command (test:tests) = catch
+  (runTest command test)
   (\case
-    ExitFailure _ -> runFailingTests command mode tests
-    ExitSuccess   -> exitSuccess)
+    ExitFailure _ -> runFailingTests command tests
+    ExitSuccess   -> exitFailure)
 
 main = do
-{-
-putStrLn "running succ L..."
-  runSucceedingTests "evaluate" (Just "L") succeedingEvalTests
-  putStrLn "running fail L"
-  runFailingTests "evaluate" (Just "L") failingEvalTests
-  putStrLn "running succ CK"
-  runSucceedingTests "evaluate" (Just "CK") succeedingEvalTests
-  putStrLn "running fail CK"
-  runFailingTests "evaluate" (Just "CK") failingEvalTests
--}
+  -- Evaluation tests
   putStrLn "running succ TCK"
-  runSucceedingTests "evaluate" "plc" succeedingEvalTests
+  runSucceedingTests (Evaluate "TCK") succeedingEvalTests
   putStrLn "running fail TCK"
-  runFailingTests "evaluate" "plc" failingEvalTests
+  runFailingTests (Evaluate "TCK") failingEvalTests
   putStrLn "running succ TCEK"
-  runSucceedingTests "evaluate" "plc" succeedingEvalTests
+  runSucceedingTests (Evaluate "TCEK") succeedingEvalTests
   putStrLn "running fail TCEK"
-  runFailingTests "evaluate" "plc" failingEvalTests
+  runFailingTests (Evaluate "TCEK") failingEvalTests
   putStrLn "running succ U..."
-  runSucceedingTests "evaluate" "uplc" succeedingEvalTests
+  runSucceedingTests (Evaluate "U") succeedingEvalTests
   putStrLn "running fail U..."
-  runFailingTests "evaluate" "uplc" failingEvalTests
-  putStrLn "running succ TC"
-  runSucceedingTests "typecheck" "plc" succeedingTCTests
+  runFailingTests (Evaluate "U") failingEvalTests
+  putStrLn "running succ TL"
+  runSucceedingTests (Evaluate "TL") succeedingEvalTests
+  putStrLn "running fail TL"
+  runFailingTests (Evaluate "TL") failingEvalTests
+
+  putStrLn "Typechecking succ"
+  runSucceedingTests Typecheck succeedingEvalTests
+-- TODO: Testing of typecheck command
```

Comment:
when will this be added?

Reply:
It's done, I should had removed the comment. 🤦

Reply:
yeah I thought you had some more tests planned!